pandas/tests/reshape/test_pivot.py

```diff
@@ -2554,3 +2554,38 @@ def test_margin_normalize(self):
             names=["A", "B"],
         )
         tm.assert_frame_equal(result, expected)
+
+    def test_pivot_with_categorical(self):
+        # gh-21370
```

Comment:
can you move this near ```def test_categorical_margins_category(self, observed)``` also if you can accept the observed fixture.

Reply:
Done

Reply:
Can you also use the ``ordered_fixture`` instead of using ``ordered=True``.

Reply:
@topper-123 just added and pushed latest changes